crates/admin/src/cluster_controller/logs_controller.rs

```diff
@@ -945,3 +988,58 @@ impl LogsController {
         }
     }
 }
+
+/// Placement hints for the [`scheduler::Scheduler`] based on the placements of logs
+#[derive(derive_more::From)]
+pub struct PartitionProcessorPlacementHints<'a> {
+    logs_controller: &'a LogsController,
+}
+
+impl<'a> scheduler::PlacementHints for PartitionProcessorPlacementHints<'a> {
+    fn preferred_nodes(&self, partition_id: &PartitionId) -> impl Iterator<Item = &PlainNodeId> {
+        let log_id = LogId::from(*partition_id);
+
+        self.logs_controller
+            .inner
+            .logs_state
+            .get(&log_id)
+            .and_then(|log_state| match log_state {
+                LogState::Available { configuration, .. } => configuration
```

Comment:
nit: maybe?
```suggestion
                LogState::Available { ref configuration, .. } => configuration
```

Reply:
I think `configuration` binds to `&Option<LogletConfiguration>` because `log_state` is a reference.